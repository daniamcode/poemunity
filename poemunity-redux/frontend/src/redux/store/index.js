import { applyMiddleware, createStore } from 'redux';
import { createLogger }                 from 'redux-logger';
import thunk                            from 'redux-thunk';
import { composeWithDevTools }          from 'redux-devtools-extension';

import rootReducer      from '../reducers';

const { NODE_ENV } = process.env


// eslint-disable-next-line import/no-mutable-exports
let configureStore;
if (NODE_ENV !== 'production') {
    const enhancer = composeWithDevTools(
        applyMiddleware(thunk, createLogger()),
    );

    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, enhancer);
    };
} else {
    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, applyMiddleware(thunk));
    };
}

export default configureStore;
