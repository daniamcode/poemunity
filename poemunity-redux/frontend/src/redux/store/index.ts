import { configureStore } from '@reduxjs/toolkit'
import {rootReducer}      from '../reducers/rootReducer';
import { useDispatch } from 'react-redux'


const store = configureStore({
    reducer: rootReducer,
  })

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch // Export a hook that can be reused to resolve types
export default store


// import { applyMiddleware, createStore } from 'redux';
// import { createLogger }                 from 'redux-logger';
// import thunk                            from 'redux-thunk';
// import { composeWithDevTools }          from 'redux-devtools-extension';

// const { NODE_ENV } = process.env

// let configureStore
// if (NODE_ENV !== 'production') {
//     const enhancer = composeWithDevTools(
//         applyMiddleware(thunk, createLogger()),
//     );

//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, enhancer);
//     };
// } else {
//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, applyMiddleware(thunk));
//     };
// }

// export default configureStore;

