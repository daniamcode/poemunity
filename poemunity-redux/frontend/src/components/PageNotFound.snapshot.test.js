import React from 'react'
import { render } from '@testing-library/react';
import PageNotFound from './PageNotFound'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux';
import store from '../redux/store';


function renderPageNotFound (arg) {
  const defaultProps = {
    match: {
      params: {}
    }
  }

  const props = { ...defaultProps, ...arg }

  return render(
    <Provider store={store}>
      <BrowserRouter>
        <PageNotFound {...props} />
      </BrowserRouter>
    </Provider>
  )
}

describe('PageNotFound', () => {
  let PageNotFoundTree

  beforeEach(async () => {
    PageNotFoundTree = renderPageNotFound()
  })

  test('should match without id', async () => {
    expect(PageNotFoundTree).toMatchSnapshot()
  })
})
