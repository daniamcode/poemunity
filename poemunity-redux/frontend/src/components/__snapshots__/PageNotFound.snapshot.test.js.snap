// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound should match without id 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="page-not-found__container"
      >
        <section
          class="page-not-found__message"
        >
          <h1
            class="page-not-found__title"
          >
            Error - 404
          </h1>
          <p
            class="page-not-found__text"
          >
            Nothing to see here
          </p>
          <a
            class="page-not-found__link"
            href="/"
          >
            Back to Dashboard
          </a>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="page-not-found__container"
    >
      <section
        class="page-not-found__message"
      >
        <h1
          class="page-not-found__title"
        >
          Error - 404
        </h1>
        <p
          class="page-not-found__text"
        >
          Nothing to see here
        </p>
        <a
          class="page-not-found__link"
          href="/"
        >
          Back to Dashboard
        </a>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
